gef.py

```diff
@@ -4308,9 +4308,42 @@ def do_invoke(self, argv):
         new_output = argv[1]
 
         disable_context()
-        res = gdb.execute("""call open("{:s}", 66, 0666)""".format(new_output), to_string=True)
-        # Output example: $1 = 3
-        new_fd = int(res.split()[2], 0)
+
+        if ':' in new_output:
+            address = socket.gethostbyname(new_output.split(':')[0])
+            port = int(new_output.split(':')[1])
+
+            # AF_INET = 2, SOCK_STREAM = 1
+            res = gdb.execute("""call socket(2, 1, 0)""", to_string=True)
+            new_fd = int(res.split()[2], 0)
+
+            # fill in memory with sockaddr_in struct contents
+            # we will do this in the stack, since connect() wants a pointer to a struct
+            vmmap = get_process_maps()
+            stack_addr = next(entry.page_start for entry in vmmap if entry.path == '[stack]')
+            original_contents = read_memory(stack_addr, 8)
+
+            write_memory(stack_addr, '\x02\x00', 2)
+            write_memory(stack_addr + 0x2, struct.pack('<H', socket.htons(port)), 2)
+            write_memory(stack_addr + 0x4, socket.inet_aton(address), 4)
+
+            info('Trying to connect to {}'.format(new_output))
+            res = gdb.execute("""call connect({}, {}, {})""".format(new_fd, stack_addr, 16), to_string=True)
+
+            # recover stack state
+            write_memory(stack_addr, original_contents, 8)
+
+            res = int(res.split()[2], 0)
+            if res == -1:
+                err('Failed to connect to {}'.format(addr))
+                return
+
+            info('Connected to {}'.format(new_output))
+        else:
+            res = gdb.execute("""call open("{:s}", 66, 0666)""".format(new_output), to_string=True)
+            # Output example: $1 = 3
```

Comment:
This comment is helpful, but let's move it up to the first time we do this result parsing.